src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -86,10 +101,8 @@ internal static bool IsCopyConstructor(Symbol member)
         internal static bool HasCopyConstructorSignature(MethodSymbol member)
         {
             NamedTypeSymbol containingType = member.ContainingType;
-            // We should relax the comparison to AllIgnoreOptions, so that a copy constructor with a custom modifier is recognized
-            // https://github.com/dotnet/roslyn/issues/45077
             return member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
-                method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
+                method.Parameters[0].Type.Equals(containingType, TypeCompareKind.AllIgnoreOptions) &&
```

Comment:
:memo: this seems more correct, although doesn't affect behavior.

Reply:
> although doesn't affect behavior.

Of course, comparison makes a difference. For example, when there is a custom modifier within the type.  

---
In reply to: [439711582](https://github.com/dotnet/roslyn/pull/45136#discussion_r439711582) [](ancestors = 439711582)